fe/fe-core/src/main/java/com/starrocks/credential/azure/AzureStorageCloudCredential.java

```diff
@@ -81,20 +84,33 @@ class AzureBlobCloudCredential extends AzureStorageCloudCredential {
 
     @Override
     void tryGenerateConfigurationMap() {
-        if (!storageAccount.isEmpty() && !sharedKey.isEmpty()) {
-            String key = String.format("fs.azure.account.key.%s.blob.core.windows.net", storageAccount);
-            generatedConfigurationMap.put(key, sharedKey);
-        } else if (!storageAccount.isEmpty() && !container.isEmpty() && !sasToken.isEmpty()) {
-            String key =
-                    String.format("fs.azure.sas.%s.%s.blob.core.windows.net", container, storageAccount);
-            generatedConfigurationMap.put(key, sasToken);
+        if (!endpoint.isEmpty()) {
+            // If user specific endpoint, they don't need to specific storage account anymore
+            // Like if user is using Azurite, they need to specific endpoint
+            if (!sharedKey.isEmpty()) {
+                String key = String.format("fs.azure.account.key.%s", endpoint);
+                generatedConfigurationMap.put(key, sharedKey);
+            } else if (!container.isEmpty() && !sasToken.isEmpty()) {
```

Comment:
If endpoint is passed, should we need to pass container? There is no parameter container in storage volume now.